plugins/measure/src/measure.cc

```diff
@@ -0,0 +1,132 @@
+ *
+ * SPDX-License-Identifier: LGPL-2.1
+ */
+
+#include "measure.hh"
+
+////////////////////////////////////////////////////////////////////////
+// Measure
+////////////////////////////////////////////////////////////////////////
+
+Measure::Measure()
+{
+}
+Measure::~Measure()
+{
+}
+
+Measure::Ptr Measure::create()
+{
+  return Ptr(new Measure());
+}
+
+////////////////////////////////////////////////////////////////////////
+// PluginInformationInterface
+////////////////////////////////////////////////////////////////////////
+
+std::string Measure::getPluginName()
+{
+  return "Measure";
+}
+
+std::string Measure::getPluginVersion()
+{
+  return "0.0";
+}
+
+void Measure::registerCapabilities(ScroomPluginInterface::Ptr host)
+{
+  host->registerViewObserver("Measure tool", shared_from_this<Measure>());
+}
+
+////////////////////////////////////////////////////////////////////////
+// ViewObserver
+////////////////////////////////////////////////////////////////////////
+
+Scroom::Bookkeeping::Token Measure::viewAdded(ViewInterface::Ptr v)
+{
+  MeasureHandler::Ptr handler = MeasureHandler::create();
+  handler->view = v;
+  v->registerSelectionListener(handler, MouseButton::SECONDARY);
+  v->registerPostRenderer(handler);
+
+  return Scroom::Bookkeeping::Token();
+}
+
+////////////////////////////////////////////////////////////////////////
+// MeasureHandler
+////////////////////////////////////////////////////////////////////////
+
+MeasureHandler::MeasureHandler()
+{
+  selection = nullptr;
+}
+MeasureHandler::~MeasureHandler()
+{
+}
+
+MeasureHandler::Ptr MeasureHandler::create()
+{
+  return Ptr(new MeasureHandler());
+}
+
+void MeasureHandler::displayMeasurement()
+{
+  std::ostringstream s;
+  s.precision(1);
+  fixed(s);
+
+  s << "l: " << selection->length()
+    << ", dx: " << selection->width()
+    << ", dy: " << selection->height()
+    << ", from: ("<< selection->start.x << "," << selection->start.y << ")"
+    << ", to: ("<< selection->end.x << "," << selection->end.y << ")";
+
+  view->setStatusMessage(s.str());
+}
+
+////////////////////////////////////////////////////////////////////////
+// SelectionListener
+////////////////////////////////////////////////////////////////////////
+
+void MeasureHandler::onSelectionStart(GdkPoint)
+{
+}
+
+void MeasureHandler::onSelectionUpdate(Selection* s)
+{
+  selection = s;
+  displayMeasurement();
+}
+
+void MeasureHandler::onSelectionEnd(Selection* s)
+{
+  selection = s;
+  displayMeasurement();
+}
+
+////////////////////////////////////////////////////////////////////////
+// PostRenderer
+////////////////////////////////////////////////////////////////////////
+
+void MeasureHandler::render(cairo_t* cr)
+{
+  if(selection)
+  {
+    GdkPoint start = view->presentationPointToWindowPoint(selection->start);
+    GdkPoint end = view->presentationPointToWindowPoint(selection->end);
+    cairo_set_line_width(cr, 1);
+    cairo_stroke(cr);
```

Comment:
You are stroking a previously set-up path? Why?

Reply:
Forgot to remove it together with the crosses it seems, removed it now.